### DISCUSSION TITLE
Golang client for lakefs

### DISCUSSION THREAD

**Discussion by tamirFireFly** [treeverse/lakeFS#4578] [Q&A]
Is there going to be golang client to lakefs?
What is the adviced way to work with lakefs currently from go ? To use just the RestAPI or there is another approach?

**Comment by eden-ohana** [comment]
Hi Tamir,
We don't publish the generated Go client code. However, we do of course generate and use it in the code. This line generates it. you could do the same https://github.com/treeverse/lakeFS/blob/141cf9bc975f295bc94c35e1a4e0fe648db0159c/pkg/api/serve.go#L3  
The CLI is a big user of the client code. look under https://github.com/treeverse/lakeFS/tree/master/cmd/lakectl/cmd
for many examples of how we use it. 
let me know if that works for you.

**Reply by tamirFireFly** [reply]
Cool thanks for the answer. 

Another question i have: there is support for writing multiple files in a bulk instead of request for each file? in the rest api for example

**Reply by eden-ohana** [reply]
No, but you can send multiple requests in parallel. Mind sharing your use case?

**Reply by tamirFireFly** [reply]
Hey im trying to store revisions of data in git based way so I can see its changes and store only the diffs so im checking if lakefs is  good for it:)

Another question: there is rest api to see lines added and removed and not just meta data of the change ?
Im see there is /repositories/{repository}/branches/{branch}/diff which let me knows which files changed but not what are the changes lines added and removed :(
I also looked at the UI it writes "Unsupported format for content diff" so what are the supported formats of data so i can get the diffs (lines added/removed)?

**Reply by eden-ohana** [reply]
Currently only markdown and Jupyter notebooks are supported on the UI.
You’re welcome to open an issue to support another format :)
The diff API will only show which file changed/added/removed without the content, the reason is that on large scale it wouldn't be scalable to read all the files content.
